app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/components/PasswordOptionAlertDialog.kt

```diff
@@ -0,0 +1,132 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.components.SwissTransferAlertDialog
+import com.infomaniak.swisstransfer.ui.components.SwissTransferTextField
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.PasswordTransferOption
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.upload.components.WeightOneSpacer
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.GetSetCallbacks
+
+@Composable
+fun PasswordOptionAlertDialog(
+    password: GetSetCallbacks<String>,
+    showPasswordOptionAlert: () -> Boolean,
+    onDismiss: () -> Unit,
+    onConfirmation: (PasswordTransferOption) -> Unit,
+    isPasswordValid: () -> Boolean,
+) {
+
+    var isPasswordActivated by rememberSaveable { mutableStateOf(password.get().isNotEmpty()) }
+    var lastValidPassword by remember { mutableStateOf("") }
+
+    fun onDismissRequest() {
+        isPasswordActivated = lastValidPassword.isNotEmpty()
+        password.set(lastValidPassword)
+        onDismiss()
+    }
+
+    fun onConfirmButtonClicked() {
+        val passwordOption = if (isPasswordActivated) {
+            PasswordTransferOption.ACTIVATED
+        } else {
+            password.set("")
+            PasswordTransferOption.NONE
+        }
+
+        lastValidPassword = password.get()
+        onConfirmation(passwordOption)
+    }
+
+    if (showPasswordOptionAlert()) {
+        SwissTransferAlertDialog(
+            titleRes = R.string.settingsOptionPassword,
+            descriptionRes = R.string.settingsPasswordDescription,
+            onDismissRequest = ::onDismissRequest,
+            onConfirmation = ::onConfirmButtonClicked,
+            shouldEnableConfirmButton = { if (isPasswordActivated) isPasswordValid() else true },
+        ) {
```

Comment:
When clicking on the password field, the `ImeAction.DONE` is missing ? We might need to add this to the SwissTransferTextField -> OutlineTextField :

```kotlin
...
keyboardOptions = KeyboardOptions(
  keyboardType = if (isPassword) KeyboardType.Password else keyboardType,
  autoCorrectEnabled = true,
  imeAction = ImeAction.Done,
),
keyboardActions = KeyboardActions(
  onDone = { /* your callback here */},
),
 ...
```

Of course, we need to make this configurable.